pilot/pkg/status/state.go

```diff
@@ -0,0 +1,251 @@
+package status
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/ghodss/yaml"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1beta1"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/utils/clock"
+
+	"istio.io/pkg/log"
+)
+
+var scope = log.RegisterScope("statusController",
+	"component for writing distribution status to istiio CRDs", 0)
+
+type Fraction struct {
+	Numerator   int
+	Denominator int
+}
+
+func (fraction Fraction) Add(fraction2 Fraction) {
+	fraction.Denominator += fraction2.Denominator
+	fraction.Numerator += fraction2.Numerator
+}
+
+type DistributionController struct {
+	lock            sync.RWMutex
+	CurrentState    map[Resource]map[string]Fraction
+	ObservationTime map[string]time.Time
+	UpdateInterval  time.Duration
+	client          dynamic.Interface
+	clock           clock.Clock
+	StaleInterval   time.Duration
+}
+
+func (c *DistributionController) Start(restConfig *rest.Config, stop <-chan struct{}) {
+	// default UpdateInterval
+	if c.UpdateInterval == 0 {
+		c.UpdateInterval = 100 * time.Millisecond
```

Comment:
This is not a polling interval, but an update interval.  This is how frequently the controller will write to status fields (note: if there are no changes since the last write, no update will be written, so 10x/second is the most frequency possible, with lower frequency probable, and no frequency at idle).